tests/unit/providers/test_anthropic_messages.py

```
@@ -133,6 +257,583 @@ async def test_amessages_non_streaming() -> None:
+                    "type": "compact_20260112",
+                    "trigger": {"type": "input_tokens", "value": 50_000},
+                }
+            ]
+        },
+    )
+
+    assert _messages_betas(params) == ["compact-2026-01-12"]
+
+
+@pytest.mark.parametrize("value", [1, 49_999, True, 50_000.0, "50000", None])
+def test_messages_betas_rejects_invalid_compaction_trigger_value(value: Any) -> None:
```

Ruff (0.16.0)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)